agent/consul/state/event_publisher.go

```diff
@@ -0,0 +1,367 @@
+package state
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-memdb"
+
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+type EventPublisher struct {
+	// topicBufferSize controls how many trailing events we keep in memory for
+	// each topic to avoid needing to snapshot again for re-connecting clients
+	// that may have missed some events. It may be zero for no buffering (the most
+	// recent event is always kept though). TODO
+	topicBufferSize int
```

Comment:
This is still not used. From some other comments it seems like it is no longer necessary to have a buffer size because we rely on GC to clean things out once the snapCacheTTL expires or the subscriptions are unsubscribed.

Should I remove this?